File: listar-topicos-reddit-java/src/main/java/com/zup/vitorhugo/listartopicosredditjava/TransformBody.java
package com.zup.vitorhugo.listartopicosredditjava;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import java.util.ArrayList;
import java.util.List;

public class TransformBody {

    public List<String> transform(String resposta) {
        List<String> titulos = new ArrayList<String>();
        try {
            final Document documento = Jsoup.parse(resposta);

            for (Element tag : documento.select(
                    "h3"
            )){
                titulos.add(tag.select("h3").text());
            }
        }catch (Exception e){
            e.printStackTrace();
        }

        return titulos;
    }

}

File: listar-topicos-reddit-kotlin/src/main/kotlin/com/zup/vitorhugo/listartopicosredditkotlin/TransformBody.kt
package com.zup.vitorhugo.listartopicosredditkotlin

import org.jsoup.Jsoup
import org.jsoup.nodes.Element
import javax.swing.text.Document

class TransformBody () {
    fun transform(resposta: String): MutableList<String> {
        var titulos = mutableListOf<String>()

        try {
            val documento: org.jsoup.nodes.Document? = Jsoup.parse(resposta)

            if (documento != null) {
                documento.select("h3").forEachIndexed { index, element ->
                    titulos.add(element.select("h3").text())
                }
            }
        }catch (e: Exception) {

        }

        return titulos
    }
}